### PR TITLE
N-api (node-addon-api)

### LINKED ISSUE #125: N-API Port & add support to node 14 & 16
### Context
vtquery is used in our APIs. Those related APIs can't be upgraded to use more recent node until this is updated.

vtquery includes c++ node bindings which should be updated to support more recent node versions.

### Notes
vtquery has old PR containing N-api port for it, which should be reviewed, tested and merged to use. This issue should be followed by other issue updating other dependencies and nodejs to more recent one.
At least vtcomposite has already been ported, as is node-cpp-skel.

Related old PR to be merged:
#120

Acceptance Criteria
vtquery uses node-addon-api instead of nan as dependency. Publish working version of vtquery binary.
vtquery supports node 14 & 16 and new binaries of vtquery are published

### PR DESCRIPTION
**Pull request by artemp**
- Nan => n-api (node-addon-api) port
- node 16 built binaries

### CODE REVIEW

**Comment by springmeyer** [comment]
Thanks @artemp! Got this in my queue to review later this week or early next.

**Comment by artemp** [comment]
> Thanks @artemp! Got this in my queue to review later this week or early next.

:+1:

File: lib/index.js [hunk 1]
```diff
@@ -40,4 +40,4 @@
  *   console.log(result); // geojson FeatureCollection
  * });
  */
-module.exports = require('./binding/vtquery.node').vtquery;
+module.exports = require('./binding/module.node').vtquery;
```
Comment (axrj) [inline lib/index.js:43]: Is there a reason for this change?

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1,2 @@
+16.14.0
```
Comment (axrj) [inline .nvmrc]: Could you pin this to the major version 16 so nvm won't download this specific version if there is another node 16.x already installed? Also helps in case there is a newer 16.x version installed on the system, that will be used instead.

Comment (OzQu) [reply .nvmrc]: Definitely!

**Review by mapsam — APPROVED** [review]
LGTM! Thanks @artemp (and apologies for the nearly 2 year delay) 

And thanks @OzQu for picking this back up!

File: lib/index.js (hunk 1 above)
Comment (artemp) [reply lib/index.js:43]: @axrj - looks like a mistake, let me check and correct.

Comment (OzQu) [reply lib/index.js:43]: I made that, although not sure was the original issue just Travis issues (which there was yesterday), and I wouldn't have needed to do that.

But in vtvalidate and vtcomponent it seems that generic name "module.node" is used, and I saw this better to be harmonized.

**Comment by OzQu** [comment]
Included bench script comparisons:

TL;DR; No significant changes. Some sections are bit faster, some slower, but no big increase/decrease on performance.

<details><summary>PR version (node 16)</summary>

```
*[n-api][~/Projects/vtquery]$ node bench/vtquery.bench.js --iterations 1000 --concurrency 5

1: pip: many building polygons ... 5814 runs/s (172ms)
2: pip: many building polygons, single layer ... 8197 runs/s (122ms)
3: query: many building polygons, single layer ... 3861 runs/s (259ms)
4: query: linestrings, mapbox streets roads ... 9804 runs/s (102ms)
5: query: polygons, mapbox streets buildings ... 5988 runs/s (167ms)
6: query: all things - dense single tile ... 2976 runs/s (336ms)
7: query: all things - dense nine tiles ... 394 runs/s (2540ms)
8: query: all things - dense nine tiles (compressed) ... 322 runs/s (3101ms)
9: elevation: terrain tile nepal ... 5263 runs/s (190ms)
10: geometry: 2000 points in a single tile, no properties ... 8772 runs/s (114ms)
11: geometry: 2000 points in a single tile, with properties ... 5128 runs/s (195ms)
12: geometry: 2000 linestrings in a single tile, no properties ... 6211 runs/s (161ms)
13: geometry: 2000 linestrings in a single tile, with properties ... 4717 runs/s (212ms)
14: geometry: 2000 polygons in a single tile, no properties ... 5291 runs/s (189ms)
15: geometry: 2000 polygons in a single tile, with properties ... 4098 runs/s (244ms)
```

</details>

<details><summary>Master version (node 14)</summary>

```
*[master][~/Projects/vtquery]$ node bench/vtquery.bench.js --iterations 1000 --concurrency 5

1: pip: many building polygons ... 6757 runs/s (148ms)
2: pip: many building polygons, single layer ... 7634 runs/s (131ms)
3: query: many building polygons, single layer ... 5376 runs/s (186ms)
4: query: linestrings, mapbox streets roads ... 9804 runs/s (102ms)
5: query: polygons, mapbox streets buildings ... 5495 runs/s (182ms)
6: query: all things - dense single tile ... 2770 runs/s (361ms)
7: query: all things - dense nine tiles ... 354 runs/s (2823ms)
8: query: all things - dense nine tiles (compressed) ... 292 runs/s (3425ms)
9: elevation: terrain tile nepal ... 5128 runs/s (195ms)
10: geometry: 2000 points in a single tile, no properties ... 8475 runs/s (118ms)
11: geometry: 2000 points in a single tile, with properties ... 5747 runs/s (174ms)
12: geometry: 2000 linestrings in a single tile, no properties ... 5747 runs/s (174ms)
13: geometry: 2000 linestrings in a single tile, with properties ... 4255 runs/s (235ms)
14: geometry: 2000 polygons in a single tile, no properties ... 4975 runs/s (201ms)
15: geometry: 2000 polygons in a single tile, with properties ... 3876 runs/s (258ms)
```

</details>
